ios/FluentUI/Shimmer/ShimmerViewSwiftUI.swift

```diff
@@ -0,0 +1,140 @@
+//
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+//
+
+import SwiftUI
+
+/// Properties that can be used to customize the appearance of the `Shimmer`.
+@objc public protocol MSFShimmerState {
+
+    /// Determines whether the shimmer is a revealing shimmer or a concealing shimmer.
+    var style: MSFShimmerStyle { get set }
+
+    /// Determines whether the view itself is shimmered or an added cover on top is shimmered
+    var shouldAddShimmeringCover: Bool { get set }
+
+    /// Whether to use the height of the view (if the view is a label), else default to token value.
+    var usesTextHeightForLabels: Bool { get set }
+}
+
+/// View Modifier that adds a "shimmering" effect to any view
+public struct ShimmerViewSwiftUI: ViewModifier, TokenizedControlView {
```

Comment:
I don't think this should be named ShimmerViewSwiftUI. I think the UIKit one needs to be renamed MSFShimmerView. Unless it's a naming collision with an existing SwiftUI control, then it should be FluentShimmerView. But I think it's the first one. And the UIKit one should probably also be demoted to just a wrapper. Though reducing that in another PR would make sense. For now, we should probably just change the name of the file and class.